kafka-connect-s3/src/main/java/io/confluent/connect/s3/auth/AwsAssumeRoleCredentialsProvider.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 Confluent Inc., Nordstrom Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.confluent.connect.s3.auth;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import io.confluent.common.config.AbstractConfig;
+import io.confluent.common.config.ConfigDef;
+import org.apache.kafka.common.Configurable;
+
+import java.util.Map;
+
+/**
+ * AWS credentials provider that uses the AWS Security Token Service to assume a Role and create a
+ * temporary, short-lived session to use for authentication.  This credentials provider does not
+ * support refreshing the credentials in a background thread.
+ */
+public class AwsAssumeRoleCredentialsProvider implements AWSCredentialsProvider, Configurable {
+
+  public static final String ROLE_EXTERNAL_ID_CONFIG = "sts.role.external.id";
+  public static final String ROLE_ARN_CONFIG = "sts.role.arn";
+  public static final String ROLE_SESSION_NAME_CONFIG = "sts.role.session.name";
+
+  private static final ConfigDef STS_CONFIG_DEF = new ConfigDef()
```

Comment:
I'd actually recommend enclosing this static definition within a method such as: 
https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java#L188

We actually might end up reusing the definitions here in the parent config.